### PR TITLE
Fix internal declaremods APIs

### PR DESCRIPTION
**Pull request by SkySkimmer**
After the parallel merge of staged declaremods and revert of removal of iter_all_segments

### CODE REVIEW

**Comment by ppedrot** [comment]
I'll merge as soon as light CI passes, if you're fine with that.

**Review by ppedrot — APPROVED** [review]

File: vernac/declaremods.ml [hunk 1]
```diff
@@ -1584,10 +1588,10 @@ let end_library ~output_native_objects dir =
 
 (** {6 Iterators} *)
 
-let iter_all_segments f =
+let iter_all_interp_segments f =
   let rec apply_obj prefix obj = match obj with
     | IncludeObject aobjs ->
-      let objs = expand_aobjs aobjs in
+      let objs = InterpVisitor.expand_aobjs aobjs in
```
Comment (SkySkimmer) [inline vernac/declaremods.ml:1594]: Originally I also did the synterp objects but search output got lots of duplicates for some reason so now it's just interp

Comment (ppedrot) [reply vernac/declaremods.ml:1594]: It's only used by Search so whatever you put in there must follow the old behaviour.

**Comment by Zimmi48** [comment]
> I'll merge as soon as light CI passes, if you're fine with that.

I wouldn't even wait for this. We already have a bunch of jobs that passed CI successfully, confirming that the problem is fixed. The rest of them are in the queue. I think this is sufficient to merge now.

**Comment by Zimmi48** [comment]
Now the light CI pipeline is fully complete. Can we merge now?

**Comment by SkySkimmer** [comment]
Yes, feel free to take the assignement from @ppedrot if you show up before him.

**Comment by Zimmi48** [comment]
@coqbot merge now

**Comment by maximedenes** [comment]
Thanks for taking care of this!

**Pull request merged**